### PR TITLE
Added restaurant show pages with reviews

### PR DESCRIPTION
**Pull request by vkk5252**
Implemented: 
- restaurant show component
- review list component
- review tile component
- reviews migration
- association between restaurants and reviews tables
- GET api endpoint for restaurant + reviews
- restaurant tiles now link to show pages

Other changes:
- changed indent spacing size to 2
- added a button to go back to the index page from the show page
- centered new restaurant form

### CODE REVIEW

**Review by Epinto150 — COMMENTED** [review]
I think the PR looks good!

File: server/src/routes/api/v1/RestaurantRouter.js [hunk 1]
```diff
@@ -17,6 +17,22 @@ restaurantRouter.get("/", async (req, res) => {
     }
 })
 
+restaurantRouter.get("/:id", async (req, res) => {
+    const { id } = req.params
+
+    try {
+        const restaurant = await Restaurant.query().findById(id)
+        restaurant.reviews = await restaurant.$relatedQuery("reviews")
```
Comment (nwalberts) [inline server/src/routes/api/v1/RestaurantRouter.js:25]: please serialize your reviews as well

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: server/src/db/seeders/ReviewSeeder.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Restaurant, Review } from "../../models/index.js";
+
+class ReviewSeeder {
+  static async seed() {
+    const chuckECheese = await Restaurant.query().findById(1)
+    const chuckECheeseReviewData = [
+      {
+        subject: "Best restaurant ever",
+        body: "adsdsa",
```
Comment (nwalberts) [inline server/src/db/seeders/ReviewSeeder.js:9]: no dummy data, replace this with an actual fake Review `body`

File: server/src/serializers/RestaurantSerializer.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  static getSummaries(restaurants) {
+    return restaurants.map(restaurant => RestaurantSerializer.getSummary(restaurant))
+  }
+
+  static async getDetails(restaurant) {
+    const serializedRestaurant = RestaurantSerializer.getSummary(restaurant)
+    const relatedReviews = await restaurant.$relatedQuery("reviews")
+    serializedRestaurant.reviews = await Promise.all(
+      relatedReviews.map(review => ReviewSerializer.getSummary(review))
```
Comment (nwalberts) [inline server/src/serializers/RestaurantSerializer.js:23]: you do not need Promise.all here

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Pull request merged**